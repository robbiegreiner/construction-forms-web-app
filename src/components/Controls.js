// eslint-disable-next-line no-unused-vars
import React, { Component } from 'react';
// eslint-disable-next-line no-unused-vars
import { Route } from 'react-router-dom';
import '../styles/Controls.css';

class Controls extends Component {

  generateButtons() {
    const buttonContent = ['Projects', 'Employees', 'All Reports', 'Reports By Project', 'Reports By Employee', 'All Forms', 'Add Employee', 'Add Project'];

    return buttonContent.map((buttonName, index) => {
      return (
        <div 
          className={ this.props.selectedControl === buttonName ? 'button selected-button' : 'button' }
          onClick={() => this.props.selectControl(buttonName)}
          key={'button-type-' + buttonName + index}
        >
          <div className={'circle-icon bg-' + index}></div>
          <p className="button-name">{buttonName}</p>
        </div>
      );
    });
  }

  render() {
    return (

      <div className="controls">
        <h1>Controls</h1>
        <hr/>
        <div className="button-container">
          {this.generateButtons()}
        </div>
      </div>

    );
  }
}

export default Controls;
