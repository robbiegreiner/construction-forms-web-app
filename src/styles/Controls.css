.controls {
  width: 50%;
  height: 100%;
  background-color: #F2F4F7;
  border-radius: 4px 0 0px 4px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.controls h1 {
  margin-top: 20px;
  margin-bottom: 20px;
}

.controls hr {
  width: 90%;
}

.button-container {
  display: flex;
  flex-wrap: wrap;
  height: 100%;
  padding: 10px;
}

.button {
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  flex-grow: 1;
  width: 33.3%;
  height: 33.3%;
  transition: background-color .3s;
  cursor: pointer;
  padding: 1px;
  box-sizing: border-box;
}

.selected-button {
  padding: 0px;
  background-color: #DEE3EB;
  border: 1px solid #1a0248;
}

.button:hover {
  background-color: #DEE3EB;
}

.bg-0 {
  background: url('../assets/projects-icon.png') no-repeat;
}

.bg-1 {
  background: url('../assets/employees-icon.png') no-repeat;
}

.bg-2 {
  background: url('../assets/form-icon.png') no-repeat;
}

.bg-3 {
  background: url('../assets/reports-project-icon.png') no-repeat;
}

.bg-4 {
  background: url('../assets/reports-employee-icon.png') no-repeat;
}

.bg-7, .bg-6 {
  background: url('../assets/plus-icon.png') no-repeat;
}

.bg-5 {
  background: url('../assets/temp-icon.png') no-repeat;
}

.circle-icon {
  background-size: 50%;
  background-position-x: center;
  background-position-y: center;
  min-height: 82px;
  width: 82px;
  border-radius: 100%;
  box-shadow: 0px 8px 24px 0px rgba(0,0,0,0.19);
  background-color: #ffffff;
}

.button-name {
  text-align: center;
}
