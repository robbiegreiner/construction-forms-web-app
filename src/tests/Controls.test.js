/* eslint-disable no-unused-vars, no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import Controls from '../components/Controls';

describe('Controls tests', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Controls />);
  });

  it('should render without crashing', () => {
    shallow(<Controls />);
  });

  it('should render buttons based on content array', () => {
    expect(wrapper.instance().generateButtons().length).toEqual(8);
  });

  it('should match snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
