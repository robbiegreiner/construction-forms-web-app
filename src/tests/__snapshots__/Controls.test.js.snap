// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls tests should match snapshot test 1`] = `
<div
  className="controls"
>
  <h1>
    Controls
  </h1>
  <hr />
  <div
    className="button-container"
  >
    <div
      className="button"
      key="button-type-Projects0"
      onClick={[Function]}
    >
      <div
        className="circle-icon bg-0"
      />
      <p
        className="button-name"
      >
        Projects
      </p>
    </div>
    <div
      className="button"
      key="button-type-Employees1"
      onClick={[Function]}
    >
      <div
        className="circle-icon bg-1"
      />
      <p
        className="button-name"
      >
        Employees
      </p>
    </div>
    <div
      className="button"
      key="button-type-All Reports2"
      onClick={[Function]}
    >
      <div
        className="circle-icon bg-2"
      />
      <p
        className="button-name"
      >
        All Reports
      </p>
    </div>
    <div
      className="button"
      key="button-type-Reports By Project3"
      onClick={[Function]}
    >
      <div
        className="circle-icon bg-3"
      />
      <p
        className="button-name"
      >
        Reports By Project
      </p>
    </div>
    <div
      className="button"
      key="button-type-Reports By Employee4"
      onClick={[Function]}
    >
      <div
        className="circle-icon bg-4"
      />
      <p
        className="button-name"
      >
        Reports By Employee
      </p>
    </div>
    <div
      className="button"
      key="button-type-All Forms5"
      onClick={[Function]}
    >
      <div
        className="circle-icon bg-5"
      />
      <p
        className="button-name"
      >
        All Forms
      </p>
    </div>
    <div
      className="button"
      key="button-type-Add Employee6"
      onClick={[Function]}
    >
      <div
        className="circle-icon bg-6"
      />
      <p
        className="button-name"
      >
        Add Employee
      </p>
    </div>
    <div
      className="button"
      key="button-type-Add Project7"
      onClick={[Function]}
    >
      <div
        className="circle-icon bg-7"
      />
      <p
        className="button-name"
      >
        Add Project
      </p>
    </div>
  </div>
</div>
`;
